### ISSUE TITLE
Use equivalent of `kubectl auth can-i` on Kube contexts to check connectivity instead of informers

### ISSUE THREAD

**Issue by feloy** [containers/podman-desktop#7629]
### Is your enhancement related to a problem? Please describe

The current implementation for checking the connectivity to Kubernetes contexts uses informers. This solution has some drawbacks:
- the informer is not able to detect that the cluster is unreachable; Podman Desktop needs to define a timeout after which the context is considered unreachable, which complicates and makes the result less accurate
- Podman Desktop doesn't know if the user has the credentials to watch the resources, but we are always watching all resources. The user having limited credentials may result in wrong information (for example for nodes, which are generally not readable in shared clusters, for example OpenShift Sandbox)

### Describe the solution you'd like

I propose to query the [`SelfSubjectRulesReview`](https://kubernetes.io/docs/reference/kubernetes-api/authorization-resources/self-subject-rules-review-v1/) resource in the cluster to check the connectivity to the cluster, before to start informers.
The advantages are:
- the request is using a POST HTTP operation, with which Podman Desktop can detect connectivity error (still using some timeout)
- the `SelfSubjectRulesReview` are expected to be accessible for all users
- Podman Desktop can use the information to start informers only on the resources the user can watch. Ultimately, we could also check the `list` permission, to have a degraded implementation using `list` operation and not `watch` (informers), if the user has `list` but not `watch` permissions.

### TODO

- [x] backend part: #10133  
- [x] dispatch to frontend: #10383
- [ ] frontend part: #10655

**Comment by feloy** [comment]
cc @deboer-tim @cdrage

**Comment by cdrage** [comment]
👍 Would really help improvement. I'm assuming this works fine with OpenShift too?

**Comment by feloy** [comment]
> 👍 Would really help improvement. I'm assuming this works fine with OpenShift too?

Yes, I have made a quick PoC, and we have access to the information on OpenShift Sandbox

**Comment by jeffmaury** [comment]
kubernetes client used groups to deal with resource manipulations but noticed we were processing Routes even on non OpenShift clusters so how do this relates ? Does it includes non standard CRDs ?

**Comment by feloy** [comment]
> kubernetes client used groups to deal with resource manipulations but noticed we were processing Routes even on non OpenShift clusters so how do this relates ? Does it includes non standard CRDs ?

Yes, the request includes CRDs, so we could be able to know if routes extension exists or not, and we could behave accordingly

**Comment by feloy** [comment]
After discussion on today's team meeting (11/07), we will use health check of cluster to check connectivity, then run this request to check the permissions, then start the informers on permitted resources.

**Comment by feloy** [comment]
Closing, as last point is covered by #10467
